api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -142,9 +142,21 @@ public static Schema selectNot(Schema schema, Set<Integer> fieldIds) {
   }
 
   public static Schema join(Schema left, Schema right) {
-    List<Types.NestedField> joinedColumns = Lists.newArrayList();
-    joinedColumns.addAll(left.columns());
```

Comment:
nit: This changes the original behavior, why not add a new function?

Reply:
@chenjunjiedada Yea that was my original version, and changed after comment of @aokolnychyi  https://github.com/apache/iceberg/pull/5376#discussion_r956441487

Reply:
Technically this is changing a public API which previously would have allowed these combos. This is ... maybe ok since it's a utility method but we may end up breaking users of the function at runtime. That said I think Anton is right and any schema with multiple columns with the same ID would always be wrong.